experimental/dds/tree2/src/test/feature-libraries/default-field-kinds/optionalChangeRebaser.spec.ts

```diff
@@ -226,36 +339,170 @@ describe("OptionalField - Rebaser Axioms", () => {
+	//   (meaning [A, B, C] ↷ D involves B ↷ compose([A⁻¹, D, A']) and C ↷ compose([B⁻¹, A⁻¹, D, A', B']))
+	const numberOfEditsToRebaseOver = 2;
+	const numberOfEditsToRebase = numberOfEditsToRebaseOver;
+
+	describe.skip("Rebase over compose exhaustive", () => {
+		for (const initialState of [{ content: undefined }, { content: "A" }]) {
+			describe(`starting with contents ${initialState.content}`, () => {
+				for (const [
+					{ description: name, changeset: edit },
+				] of generatePossibleSequenceOfEdits(
+					initialState,
+					generateChildStates,
+					1,
+					"local-rev-",
+				)) {
+					for (const namedEditsToRebaseOver of generatePossibleSequenceOfEdits(
+						initialState,
+						generateChildStates,
+						numberOfEditsToRebaseOver,
+						"trunk-rev-",
+					)) {
+						const title = `Rebase ${name} over compose ${JSON.stringify(
+							namedEditsToRebaseOver.map(({ description }) => description),
+						)}`;
+
+						it(title, () => {
+							const editsToRebaseOver = namedEditsToRebaseOver.map(
+								({ changeset }) => changeset,
+							);
+							const rebaseWithoutCompose = rebaseTagged(edit, ...editsToRebaseOver);
+							const metadata = defaultRevisionMetadataFromChanges([
+								...editsToRebaseOver,
+								edit,
+							]);
+							const rebaseWithCompose = rebaseComposed(
+								metadata,
+								edit,
+								...editsToRebaseOver,
+							);
+							assert.deepEqual(rebaseWithCompose.change, rebaseWithoutCompose.change);
+						});
+					}
+				}
+			});
+		}
+	});
+
+	describe.skip("Sandwich rebase over compose exhaustive", () => {
+		for (const initialState of [{ content: undefined }, { content: "A" }]) {
+			describe(`starting with contents ${initialState.content}`, () => {
+				for (const namedSourceEdits of generatePossibleSequenceOfEdits(
+					initialState,
+					generateChildStates,
+					numberOfEditsToRebase,
+					"local-rev-",
+				)) {
+					for (const [
+						{ description: name, changeset: namedEditToRebaseOver },
+					] of generatePossibleSequenceOfEdits(
+						initialState,
+						generateChildStates,
+						1,
+						"trunk-rev-",
+					)) {
+						const title = `Rebase ${JSON.stringify(
+							namedSourceEdits.map(({ description }) => description),
+						)} over ${name}`;
+
+						it(title, () => {
+							const editToRebaseOver = namedEditToRebaseOver;
+							const sourceEdits = namedSourceEdits.map(({ changeset }) => changeset);
+
+							const inverses = sourceEdits.map((change) =>
+								tagRollbackInverse(
+									invert(change),
+									`rollback-${change.revision}` as RevisionTag,
+									change.revision,
+								),
+							);
+
+							const rebasedEditsWithoutCompose: TaggedChange<OptionalChangeset>[] =
+								[];
+							const rebasedEditsWithCompose: TaggedChange<OptionalChangeset>[] = [];
+
+							for (let i = 0; i < sourceEdits.length; i++) {
+								const edit = sourceEdits[i];
+								const editsToRebaseOver = [
+									...inverses.slice(0, i),
```

Comment:
Shouldn't this list be reversed?

Reply:
Probably best to reverse `inverses` after it is created, so that `const allTaggedEdits = [...inverses, editToRebaseOver];` also benefits from the correct ordering.

Reply:
thank you, good catch!